util/src/main/java/tech/pegasys/artemis/util/async/AsyncRunnerProd.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.util.async;
+
+import static tech.pegasys.artemis.util.async.SafeFuture.propagateResult;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+public class AsyncRunnerProd implements AsyncRunner {
```

Comment:
I'd call this `DelayedExecutorAsyncRunner` - the name relates to what makes it's implementation special (e.g. `ArrayList` because it uses an array backing).

Reply:
That makes sense to me. Done.